### DISCUSSION TITLE
How to change 2FA Email address?

### DISCUSSION THREAD

**Discussion by cwdtle** [dani-garcia/vaultwarden#3232] [Q&A] [answered]
Hi,
unfortunately we need to change the mail addresses of all users.
The users are scattered around multiple organisations.
2FA is required via policy in all organisations.
Via "account setting -> Security -> Two-step login -> Email -> Manage" the only option is to turn off Email 2FA.
However turning off 2FA, will remove the user from all organisations requiring 2FA via policy.
So is there any way to change the 2FA mail addresses without beeing removed from all organisations?

**Comment by jjlin** [comment, marked as answer]
The simplest way is probably to update the database directly. 2FA info is stored in the `twofactor` table, and email 2FA entries are of the form
```
uuid|user_uuid|atype|enabled|data|last_used
-------------------------------------------
<some-uuid>|<some-uuid>|1|1|{"email":"user@domain.com","last_token":null,"token_sent":<integer>,"attempts":0}|0
```
So you would just look for 2FA entries where `atype` is 1, then read out the `data` field, update the email address, and write it back.

Or if you're not very familiar with SQL, you could just dump the entire database, edit in a text editor, and load it back.

**Reply by cwdtle** [reply]
Great, thanks!
